### ISSUE TITLE
Trezor Suite Mac Doesn't display BTC recent transactions

### ISSUE THREAD

**Issue by yunesco** [trezor/trezor-suite#3671]
Trezor Suite Mac Doesn't display BTC recent transactions but Trezor Suite Web does. See screenshots attached.

Mac Trezor Suite
<img width="1085" alt="Screenshot 2021-04-22 at 11 39 21" src="https://user-images.githubusercontent.com/12301590/115692825-9d3a5b00-a35f-11eb-886b-80c5a7e7e616.png">

Web Trezor Suite
<img width="978" alt="Screenshot 2021-04-22 at 11 39 10" src="https://user-images.githubusercontent.com/12301590/115692817-9b709780-a35f-11eb-8bcf-5fe1cf5b6ddc.png">

I was able to solve this problem by ejecting the wallet and then reimporting it.

**Comment by keraf** [comment]
Hey there! 

Could you provide us some logs from the desktop app? It might give us a better idea what's happening. To do so, run Suite Desktop in command line using the `--log-level=debug` parameter as well as `--log-write` (to have it written in your home directory). 

I can only guess what happened and I would say maybe it might be related to Tor? Do you have it enabled? 

@trezor/qa Did you guys ever encounter a similar issue?

**Comment by bosomt** [comment]
@keraf nope, never

**Comment by bosomt** [comment]
I would guess it was typo in passphrase :)

**Comment by yunesco** [comment]
Passhrase is correct, i can see the other coins, just not BTC :)

**Comment by yunesco** [comment]
@keraf I did have Tor enabled, unfortunately, I can't give you the logs because I installed the app via the .dmg file not source code

**Comment by slowbackspace** [comment]
Since ejecting the wallet helped I would guess that account got saved in some weird broken state. It shouldn't even be a problem related to fetching the data. From the screenshot it looks like data for the account were already fetched that means we already should have last 25 txs (we are fetching basic account details together with last 25 txs).

The screen @Yunislav sees is used basically a fallback for XRP transactions, where we can tell the account is not empty, yet the backend doesn't return any transactions because it prunes old data. You are not supposed to see this screen in other coins. For every coin except XRP (maybe eth where your only tx is internal one), an empty account should have 0 transactions and non empty should return at least one transaction.
This is some weird state where account is not empty and has no transactions stored in reducer at the same time.

~~The only way how to replicate this I can think of is to remember your account and then delete object store (or just items) responsible for storing transactions from indexedDB. Then on next launch suite will retrieve data from indexedDB: non empty account and 0 txs for this account. So basically either something went wrong while saving the transactions or retrieving them from DB or the data got corrupted while being stored.~~

~~We could implement one workaround for such cases and that is validate the data retrieved from indexedDB, if we see account.empty set to false and yet there are no stored txs we can tell the data is wrong and we need to refetch them. But there might be more serious underlying problem causing the issue in the first place.~~

edit: I replicated the issue few comments below, most likely related to account update mechanism.

@Yunislav I assume you cannot reproduce the bug after ejecting the wallet and rediscovering your accounts again, is that correct?

**Comment by yunesco** [comment]
@slowbackspace Nope i couldn't reproduce the bug. When i sent the transaction to the BTC wallet, i could see a notification that i received BTC to my address but i couldn't see it on the transaction history.

**Comment by slowbackspace** [comment]
@Yunislav any chance that the bug appeared only after you auto upgraded from older release? We've recently released 21.4.1 that was pushed as semi-automatic update (you got a popup in the old Suite asking you to upgrade). I am wondering if it could be messed up data migration during the update.

Another weird thing is that this happens only for some accounts https://github.com/trezor/trezor-suite/issues/3650. So it is not botched auto-upgrade process, but maybe still something wrong in db migration.

edit: okay it was fresh install, no chance the bug is in db migration/updater.

**Comment by yunesco** [comment]
@slowbackspace This was the first time i installed Trezor Suite on this computer, so i don't think so

**Comment by slowbackspace** [comment]
Thank you for your help!
I replicated the issue.

Steps:
1. In desktop app create new wallet (I picked some random passphrase) with empty bech32 account (testnet is fine and looks like existing passphrase is also fine, but I am not sure) 
2. Remember the wallet
3. Open web app and send transaction to the fresh account in desktop app
4. You will get pending tx on both the web and desktop app. Close the desktop app
5. Wait till tx is confirmed and then open desktop app again
6. ???
7. Profit (are we still doing this?)

tl;dr: when you close the app while the tx is pending then on the next launch all txs are removed, probably because we find out that we have incorrect number of txs and need to refetch the data from blockchain, but it seems it fails to fetch new data.
This has probably something to do with a way we update account data on new block. Some optimisations were done in https://github.com/trezor/trezor-suite/issues/3169, not sure if it could get broken that long ago, but worth mentioning.

cc @trezor/qa Could you confirm?

**Comment by sorooris** [comment]
@slowbackspace Oui monsieur, confirming the issue according to your steps!

Suite: app 21.4.1 1baa8ce
OS: macOS Big Sur 11.3
Device: T1 1.10.0

Screenshot:
![image](https://user-images.githubusercontent.com/10185314/117625059-e8cd7100-b175-11eb-8279-7d1374b379cf.png)
Log:
```Qs-MacBook-Air:~ qa$ /Applications/Trezor\ Suite.app/Contents/MacOS/Trezor\ Suite --log-path=/Users/qa/Documents/Suite\ logs --log-file=macos_log.txt --log-level=debug --log-write --pre-release
2021-05-10T07:47:49.192Z - INFO(main): Application starting
2021-05-10T07:47:49.213Z - INFO(main): Application starting
2021-05-10T07:47:49.351Z - INFO(build): Info:
2021-05-10T07:47:49.351Z - INFO(build): - Version: 21.4.1
2021-05-10T07:47:49.352Z - INFO(build): - Commit: 1baa8ced16f2f6a5d60e9f4d7911669f856123d8
2021-05-10T07:47:49.353Z - INFO(build): - Dev: No
2021-05-10T07:47:49.353Z - INFO(build): - Args: --log-path=/Users/qa/Documents/Suite logs --log-file=macos_log.txt --log-level=debug --log-write --pre-release
2021-05-10T07:47:49.353Z - INFO(build): - CWD: /Users/qa
2021-05-10T07:47:53.317Z - DEBUG(computer): Info:
2021-05-10T07:47:53.319Z - DEBUG(computer): - Platform: darwin x64
2021-05-10T07:47:53.319Z - DEBUG(computer): - OS: macOS 11.3
2021-05-10T07:47:53.320Z - DEBUG(computer): - Manufacturer: Apple Inc.
2021-05-10T07:47:53.321Z - DEBUG(computer): - Model: MacBookAir7,2
2021-05-10T07:47:53.322Z - DEBUG(computer): - Virtual: No
2021-05-10T07:47:53.323Z - DEBUG(computer): - CPU: Intel® Core™ i5-5350U
2021-05-10T07:47:53.323Z - DEBUG(computer): - Cores: 1x2(+2) @ 1.8GHz
2021-05-10T07:47:53.327Z - DEBUG(computer): - RAM: 8.0 GB
2021-05-10T07:47:53.337Z - DEBUG(init): Create Browswer Window (1087x700)
2021-05-10T07:47:53.523Z - DEBUG(init): Load URL (file:///index.html)
2021-05-10T07:47:53.561Z - INFO(modules): Loading 13 modules
2021-05-10T07:47:53.562Z - DEBUG(modules): Loading event-logging/process
2021-05-10T07:47:53.572Z - DEBUG(modules): Loading event-logging/app
2021-05-10T07:47:53.572Z - DEBUG(modules): Loading event-logging/contents
2021-05-10T07:47:53.572Z - DEBUG(modules): Loading menu
2021-05-10T07:47:53.572Z - DEBUG(modules): Loading shortcuts
2021-05-10T07:47:53.572Z - DEBUG(modules): Loading request-filter
2021-05-10T07:47:53.572Z - DEBUG(modules): Loading external-links
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading window-controls
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading http-receiver
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading metadata
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading bridge
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading tor
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading analytics
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading csp
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading file-protocol
2021-05-10T07:47:53.573Z - DEBUG(modules): Loading auto-updater
2021-05-10T07:47:53.777Z - INFO(http-receiver): Starting server
2021-05-10T07:47:53.781Z - INFO(bridge): Starting (Dev: No)
2021-05-10T07:47:53.871Z - INFO(auto-updater): Is looking for pre-releases? (Yes)
2021-05-10T07:47:53.871Z - INFO(modules): All modules loaded
2021-05-10T07:47:53.885Z - INFO(http-receiver): Server started
2021-05-10T07:47:54.068Z - ERROR(process-trezord): Status error: request to http://127.0.0.1:21325/ failed, reason: connect ECONNREFUSED 127.0.0.1:21325
2021-05-10T07:47:54.069Z - DEBUG(process-trezord): Setting a restart throttle (3)
2021-05-10T07:47:54.069Z - INFO(process-trezord): Starting process:
2021-05-10T07:47:54.069Z - INFO(process-trezord): - Path: /Applications/Trezor Suite.app/Contents/Resources/bin/bridge/trezord
2021-05-10T07:47:54.069Z - INFO(process-trezord): - Params: 
2021-05-10T07:47:54.069Z - INFO(process-trezord): - CWD: /Applications/Trezor Suite.app/Contents/Resources/bin/bridge
2021-05-10T07:47:54.458Z - DEBUG(window-control): Focus
2021-05-10T07:47:57.070Z - DEBUG(process-trezord): Clearing throttle
2021-05-10T07:47:57.727Z - DEBUG(window-control): Client ready
2021-05-10T07:47:57.990Z - DEBUG(tor): Getting status (OFF)
2021-05-10T07:47:58.108Z - INFO(auto-updater): Update checking request (manual: No)
2021-05-10T07:47:58.109Z - INFO(auto-updater): Checking for update
2021-05-10T07:47:58.566Z - INFO(analytics): Getting OS version
2021-05-10T07:47:59.314Z - DEBUG(csp): Header applied to https://data.trezor.io/suite/log/desktop/beta.log?c_v=1.6&c_type=desktop-init&c_commit=1baa8ced16f2f6a5d60e9f4d7911669f856123d8&c_instance_id=xXfuy8JbUc&c_session_id=f8UVDgvOq2&desktopOSVersion=darwin_20.4.0
2021-05-10T07:47:59.317Z - DEBUG(csp): Header applied to https://data.trezor.io/suite/log/desktop/beta.log?c_v=1.6&c_type=suite-ready&c_commit=1baa8ced16f2f6a5d60e9f4d7911669f856123d8&language=en&enabledNetworks=btc%2Cltc%2Ceth&localCurrency=usd&discreetMode=false&screenWidth=1440&screenHeight=900&platform=MacIntel&platformLanguage=en-GB&tor=false&rememberedStandardWallets=0&rememberedHiddenWallets=1&theme=dark&suiteVersion=
2021-05-10T07:47:59.642Z - INFO(auto-updater): No new update is available:
2021-05-10T07:47:59.642Z - INFO(auto-updater): - Last version: 21.4.1
2021-05-10T07:47:59.642Z - INFO(auto-updater): - Last release date: 2021-04-14T07:46:37.944Z
2021-05-10T07:47:59.642Z - INFO(auto-updater): - Manual check: No
2021-05-10T07:47:59.694Z - DEBUG(bridge): Setting header for http://127.0.0.1:21325/
2021-05-10T07:47:59.697Z - DEBUG(csp): Header applied to http://127.0.0.1:21325/
2021-05-10T07:47:59.982Z - DEBUG(bridge): Setting header for http://127.0.0.1:21325/enumerate
2021-05-10T07:47:59.988Z - DEBUG(csp): Header applied to http://127.0.0.1:21325/enumerate
2021-05-10T07:48:00.193Z - DEBUG(bridge): Setting header for http://127.0.0.1:21325/listen
2021-05-10T07:48:00.290Z - DEBUG(csp): Header applied to https://data.trezor.io/suite/log/desktop/beta.log?c_v=1.6&c_type=transport-type&c_commit=1baa8ced16f2f6a5d60e9f4d7911669f856123d8&c_instance_id=xXfuy8JbUc&c_session_id=f8UVDgvOq2&type=bridge&version=2.0.31
2021-05-10T07:48:01.446Z - DEBUG(csp): Header applied to wss://eth2.trezor.io/websocket
2021-05-10T07:48:01.449Z - DEBUG(csp): Header applied to wss://ltc1.trezor.io/websocket
2021-05-10T07:48:01.557Z - DEBUG(csp): Header applied to wss://btc1.trezor.io/websocket
2021-05-10T07:48:01.777Z - DEBUG(csp): Header applied to https://data.trezor.io/suite/log/desktop/beta.log?c_v=1.6&c_type=router%2Flocation-change&c_commit=1baa8ced16f2f6a5d60e9f4d7911669f856123d8&c_instance_id=xXfuy8JbUc&c_session_id=f8UVDgvOq2&prevRouterUrl=%2F&nextRouterUrl=%2F
2021-05-10T07:48:02.034Z - DEBUG(csp): Header applied to https://news.trezor.io/posts?limit=4
2021-05-10T07:48:02.088Z - DEBUG(csp): Header applied to https://cdn.trezor.io/static/medium/images/max/1024/1*CbKJXPhG6yaU2pYrgVL9uw.png
2021-05-10T07:48:02.091Z - DEBUG(csp): Header applied to https://cdn.trezor.io/static/medium/images/max/1024/1*d8SCcQKa7Pmv4iuzqX-ezA.png
2021-05-10T07:48:02.154Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0x79186Ba0FC6fa49fd9DB2F0bA34F36F8c24489c7&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.162Z - DEBUG(csp): Header applied to https://cdn.trezor.io/static/medium/images/max/1024/1*A5FVKtN8Wtx7Mqc9Qkt9hw.png
2021-05-10T07:48:02.184Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0x519475b31653E46D20cD09F9FdcF3B12BDAcB4f5&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.194Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0x514910771AF9Ca656af840dff83E8264EcF986CA&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.195Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0x58b6A8A3302369DAEc383334672404Ee733aB239&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.196Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0x8E870D67F660D95d5be530380D0eC0bd388289E1&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.197Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0x0000000000085d4780B73119b644AE5ecd22b376&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.198Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0xdAC17F958D2ee523a2206206994597C13D831ec7&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:02.230Z - DEBUG(csp): Header applied to https://cdn.trezor.io/dynamic/coingecko/api/v3/simple/token_price/ethereum?contract_addresses=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48&vs_currencies=usd,eur,gbp,aed,ars,aud,bdt,bhd,bmd,brl,cad,chf,clp,cny,czk,dkk,hkd,huf,idr,ils,inr,jpy,krw,kwd,lkr,mmk,mxn,myr,nok,nzd,php,pkr,pln,rub,sar,sek,sgd,thb,try,twd,vef,vnd,zar,xdr,xag,xau
2021-05-10T07:48:04.637Z - DEBUG(csp): Header applied to https://data.trezor.io/suite/log/desktop/beta.log?c_v=1.6&c_type=menu%2Fgoto%2Fwallet-index&c_commit=1baa8ced16f2f6a5d60e9f4d7911669f856123d8&c_instance_id=xXfuy8JbUc&c_session_id=f8UVDgvOq2
2021-05-10T07:48:04.782Z - DEBUG(csp): Header applied to https://data.trezor.io/suite/log/desktop/beta.log?c_v=1.6&c_type=router%2Flocation-change&c_commit=1baa8ced16f2f6a5d60e9f4d7911669f856123d8&c_instance_id=xXfuy8JbUc&c_session_id=f8UVDgvOq2&prevRouterUrl=%2F&nextRouterUrl=%2Faccounts%2F
2021-05-10T07:48:19.091Z - DEBUG(window-control): Blur
2021-05-10T07:48:25.379Z - DEBUG(window-control): Focus
2021-05-10T07:48:27.713Z - DEBUG(window-control): Blur
^C2021-05-10T07:48:29.213Z - INFO(app): Quitting
2021-05-10T07:48:29.237Z - INFO(http-receiver): Stopping server (app quit)
2021-05-10T07:48:29.238Z - INFO(tor): Stopping (app quit)
2021-05-10T07:48:29.239Z - WARN(process-tor): Couldn't stop process (already stopped)
2021-05-10T07:48:29.239Z - INFO(bridge): Stopping (app quit)
2021-05-10T07:48:29.239Z - INFO(process-trezord): Stopping process
2021-05-10T07:48:29.240Z - INFO(http-receiver): Server stopped
2021-05-10T07:48:29.345Z - ERROR(exception): write after end
2021-05-10T07:48:29.347Z - INFO(process-trezord): Exited, code: N/A (Stopped: Yes)

**Comment by uj** [comment]
Is this possibly related to https://github.com/trezor/trezor-suite/issues/3848 ?

**Comment by bosomt** [comment]
I was able to reproduce issue with disconnected device.See video.
https://www.youtube.com/watch?v=rXOdAqRVPws&feature=youtu.be

![image](https://user-images.githubusercontent.com/31506317/126612507-36ba9557-c057-4d75-951e-9145a23473e5.png)

**Steps to reproduce:**
1) remember first wallet and disconnect it
2) disable all coins
3) connect different device
4) enable some coin
5) switch to saved wallet
6) observe

**Info:**
 - Suite version: web 21.7.1 (24cc565a3badd85ab48b30348022471dfd85f90e)
 - Browser: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:90.0) Gecko/20100101 Firefox/90.0
 - OS: Linux x86_64
 - Screen: 2560x1440
 - Device: model T 2.4.2 regular

**Comment by bosomt** [comment]
`Account does not exist` problem is still present :( 
looks like different issue.

<img width="982" alt="obrazek" src="https://user-images.githubusercontent.com/31506317/140190956-93609f30-f915-48e7-9f80-7574d09349c3.png">

**Info:**
 - Suite version: desktop 21.11.0 (4a56ae794159091a4f2c4de847d237a47657dc1c)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/21.11.0 Chrome/93.0.4577.58 Electron/14.0.0 Safari/537.36
 - OS: MacIntel
 - Screen: 1680x1050
 - Device: model 1 1.10.3 regular

**Comment by tomasklim** [comment]
> `Account does not exist` problem is still present :( looks like different issue.
> 
> <img alt="obrazek" width="982" src="https://user-images.githubusercontent.com/31506317/140190956-93609f30-f915-48e7-9f80-7574d09349c3.png">
> 
> **Info:**
> 
> * Suite version: desktop 21.11.0 ([4a56ae7](https://github.com/trezor/trezor-suite/commit/4a56ae794159091a4f2c4de847d237a47657dc1c))
> * Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/21.11.0 Chrome/93.0.4577.58 Electron/14.0.0 Safari/537.36
> * OS: MacIntel
> * Screen: 1680x1050
> * Device: model 1 1.10.3 regular

Is this the problem with two devices? If yes, please create a new issue for it. 

The problem this issue is talking about should be fixed.

**Comment by bosomt** [comment]
@tomasklim it was just one T1 device and only Standard wallet was loaded
Suite storage was wiped less than hour before.
As soon i will reproduce it again i will send proper issue

**Comment by bosomt** [comment]
created separate issue, it happened to me twice in last 24 hours.
https://github.com/trezor/trezor-suite/issues/4465
